docs/basic-features/environment-variables.md

```diff
@@ -0,0 +1,108 @@
+
+You can use the shell or any other tool that runs before the [Next.js CLI](/api-reference/cli) to add environment variables. For example, using the bash:
+
+```bash
+NEXT_PUBLIC_EXAMPLE_KEY=my-value next dev
+```
+
+And using [cross-env](https://github.com/kentcdodds/cross-env):
+
+```bash
+npx cross-env NEXT_PUBLIC_EXAMPLE_KEY=my-value next dev
+```
+
+### Caveats
+
+- Trying to destructure `process.env` variables won't work due to the nature of webpack [DefinePlugin](https://webpack.js.org/plugins/define-plugin/).
+- When dealing with secrets, it's better to [expose the variables using `.env`](#exposing-environment-variables), to avoid exposing secrets in your build output
+
+## Exposing Environment Variables
+
+Next.js allows you to expose variables using an environment variables file (`.env`), with included support for multiple environments, it works like this:
+
+- `.env` - Contains environment variables for all environments
+- `.env.local` - Local variable overrides for all environments
+- `.env.[environment]` - Environment variables for one environment. For example: `.env.development`
+- `.env.[environment].local` - Local variable overrides for one environment. For example: `.env.development.local`
+
+> **Note**: `.env` files **should be** included in your repository, but **`.env*.local` files should be ignored**. Consider `.local` files as a good place for secrets, and non-local files as a good place for defaults.
+
+The supported environments are `development`, `production` and `test`. The environment is selected in the following way:
+
+- [`next dev`](/docs/api-reference/cli#development) uses `development`
+- [`next build`](/docs/api-reference/cli#build) and [`next-start`](/docs/api-reference/cli#production) use `production`
+- `NODE_ENV=test next dev` will use `test` instead of the default.
+
+If the same environment variable is defined multiple times, the priority of which variable to use is decided in the following order:
+
+- Already defined environment variables have the higher priority. For example: `MY_KEY=value next dev`
+- `.env.[environment].local`
+- `.env.[environment]`
+- `.env.local`
+- `.env`
+
+For example, consider the file `.env.local` with the following content:
+
+```bash
+API_KEY='my-secret-api-key'
+NEXT_PUBLIC_APP_LOCALE='en-us'
+```
+
+And the following page:
+
+```jsx
+export default function Page() {
+  return <h1>The locale is set to: {process.env.NEXT_PUBLIC_APP_LOCALE}</h1>
+}
+
+export async function getStaticProps() {
+  const db = await myDB(process.env.API_KEY)
+  // ...
+}
+```
+
+`process.env.NEXT_PUBLIC_APP_LOCALE` will be replaced with `'en-us'` in the build output. Because variables that start with `NEXT_PUBLIC_` will be [inlined at build time](#inlined-environment-variables).
+
+`process.env.API_KEY` will be a variable with `'my-secret-api-key'` at build time and runtime, but the build output will not have access to the key.
```

Comment:
> but the build output will not have access to the key

You might want to explain why this is the case because it's not immediately obvious. 

Maybe something like:

```suggestion
`process.env.API_KEY` will be a variable with `'my-secret-api-key'` at build time and runtime, but the build output will not contain this key. This is because `process.env.API_KEY` is only used by `getStaticProps` which runs only on the server — and only the *result* of running `getStaticProps` is included in the build output.
```